### PR TITLE
Update certificates button to not render when not having template

### LINKED ISSUE #275: View Certificates button should not show up when no certificate template is selected
### Steps to Reproduce

* Ensure the _Course Completed Page_ setting has been configured.
* Add the _Course Completed Actions_ block to the _Course Completed_ page.
* Ensure the _View Certificate_ block appears.
* Create a course, and keep the Certificate Template as "none".
* Complete the course to be redirected to _Course Completed_ page.

### What I Expected

To not show up have the View Certificates button.

### What Happened Instead

The View Certificates button appears.

### PHP / WordPress / Sensei Certificates / Sensei LMS version

Sensei Certificates 2.2.0 / Sensei LMS 3.13.0

### Screenshot / Video

https://user-images.githubusercontent.com/876340/129405376-2eb90249-7704-4b8b-8668-675ffb201a3e.mov

### Context

p1628781354120600-slack-C8QJAM52B

### PR DESCRIPTION
**Pull request by renatho**
Fixes #275

### Changes proposed in this Pull Request

* It updates the certificates button in the frontend to not render when not having a template.

### Testing instructions

* Set up the Course Completed page.
* Create a course without a Certificate Template.
* Complete the course.
* Go to the Results page, and make sure you don't see the "View Certificate" button.
* Update the course, adding a Certificate Template, and make sure the button shows up.

### CODE REVIEW

File: classes/class-woothemes-sensei-certificates.php [hunk 1]
```diff
@@ -1483,24 +1483,36 @@ public function enqueue_block_editor_assets() {
...
-		// Check that the course ID exists and that the user has completed the course.
 		if (
+			// Check that the course ID exists and that the user has completed the course.
 			! $course_id
 			|| ! get_current_user_id()
 			|| 'course' !== get_post_type( $course_id )
 			|| ! Sensei_Utils::user_completed_course( $course_id, get_current_user_id() )
+
+			// Check that the block is a core/button and it contains the respective class name.
+			|| ! isset( $block['blockName'] )
```
Comment (renatho) [inline classes/class-woothemes-sensei-certificates.php:1499]: This check is also in `update_button_block_url`, but I needed to test it earlier to return the block when it's not the respective block, and empty when it doesn't have the template.

File: classes/class-woothemes-sensei-certificates.php [hunk 2]
```diff
@@ -1483,24 +1483,36 @@ public function enqueue_block_editor_assets() {
...
+			|| 'core/button' !== $block['blockName']
+			|| ! isset( $block['attrs']['className'] )
+			|| false === strpos( $block['attrs']['className'], $class_name )
 		) {
 			return $block_content;
 		}
 
-		if ( ! method_exists( 'Sensei_Blocks', 'update_button_block_url' ) ) {
```
Comment (renatho) [inline classes/class-woothemes-sensei-certificates.php:1499]: I think it makes more sense to return an empty block in this case, so I changed this.

File: classes/class-woothemes-sensei-certificates.php [hunk 3]
```diff
@@ -1483,24 +1483,36 @@ public function enqueue_block_editor_assets() {
...
 			! $course_id
 			|| ! get_current_user_id()
 			|| 'course' !== get_post_type( $course_id )
 			|| ! Sensei_Utils::user_completed_course( $course_id, get_current_user_id() )
+
+			// Check that the block is a core/button and it contains the respective class name.
+			|| ! isset( $block['blockName'] )
+			|| 'core/button' !== $block['blockName']
+			|| ! isset( $block['attrs']['className'] )
+			|| false === strpos( $block['attrs']['className'], $class_name )
```
Comment (alexsanford) [inline classes/class-woothemes-sensei-certificates.php:1502]: Technically, this is not an exact match. For example, a button with the class `should-view-certificates` is matched here, and follows the same logic. This also applies to the identical logic in `update_button_block_url`.

An exact match using a regex would be tighter. I'm not sure it's super important though 🤷‍♂️

File: classes/class-woothemes-sensei-certificates.php (hunk 1 above)
Comment (alexsanford) [reply classes/class-woothemes-sensei-certificates.php:1499]: Would it be worth extracting these checks to a method that could be used in both places?

**Review by alexsanford — COMMENTED** [review]
Works well! A couple small comments below.

File: classes/class-woothemes-sensei-certificates.php (hunk 3 above)
Comment (renatho) [reply classes/class-woothemes-sensei-certificates.php:1502]: That's true! We're using it in many different cases of the code (even in the Sensei core). I think we could keep it as it is for now, and refactor all the cases to use regex if we have any user with problems. We could create a single helper function in the core for that instead replicate the same regex in many different places.
WDYT? If you think it's important, I can do it, but I'd do it in a different PR to handle all the cases.

File: classes/class-woothemes-sensei-certificates.php (hunk 1 above)
Comment (renatho) [reply classes/class-woothemes-sensei-certificates.php:1499]: Since it's a method from the core, we'd need to write this helper method there, and also replicate it here in case their core plugin is not with the version with the helper function. So I don't think it's worth doing now. WDYT?

Comment (alexsanford) [reply classes/class-woothemes-sensei-certificates.php:1499]: Right, gotcha. Ok that's fine by me 👍

**Review by alexsanford — APPROVED** [review]
